[Sim Kian Seng] iP

simkianseng opened the pull request:
# Simpli
> "The more you simp~lify~ the easier your life." - Sim Kian Seng

The ultimate simping tool that simp-lies your life :smile:.
&nbsp;

Tired of the plain old `System.out.println("Hello blah blah blah")` greeting message? Check out this new chatbot's greeting message. This chatbot greets you with zest, here is a sneak peek of its greeting program:
```java
public void greet() {
    ui.display(
            "Hello! I'm SIMP-LI\n" +
                    "How can I simp-lify your life?"
    );
}
```
Wow! Sick program right?!?!?!?!?!?!? But wait there's more...
&nbsp;

Tired of plain old checkboxes to checkout your tasks like the ones below?
- [x] Task 1
- [ ] Task 2
- [ ] Task 3

Fred not, just use this chatbot.
&nbsp;


The chatbot can:
- **Add** todo, deadline and event tasks
- **Mark** and **unmark** tasks as completed and uncompleted respectively
- **Save** and **load** tasks to and from the filesystem
- **Search** tasks by a keyword
&nbsp;

You can use it in 3 easy steps:
1. _Type_ your commands
2. _Enter_ key
3. _View_ the output
&nbsp;

What are you waiting for? View the project [here](https://github.com/SimKianSeng/ip).

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

simkianseng committed:
Add chatbot rename, greet and exit messages

simkianseng committed:
Add echo

simkianseng committed:
Add store user input

simkianseng committed:
Add mark as done

simkianseng committed:
Update Duke.java

simkianseng committed:
Add todos, events, deadlines

simkianseng committed:
Add text ui testing

simkianseng committed:
Add Interpreter.java

simkianseng committed:
Update runtest.bat

simkianseng committed:
Add error handling

simkianseng committed:
Add delete task

simkianseng committed:
Update project structure

simkianseng committed:
Refactor overall code

simkianseng committed:
Add Action.java enum

simkianseng committed:
Update Interpreter.java

simkianseng committed:
Add task saving and loading

simkianseng committed:
Merge branch 'branch-level-7'

simkianseng committed:
Add string interpretation as LocalDateTime object

simkianseng committed:
Merge branch 'branch-level-8'

simkianseng committed:
Add Storage.java TaskManager.java Ui.java

simkianseng committed:
Merge branch 'branch-A-MoreOOP'

simkianseng committed:
Move multiple Java class files into various packages

simkianseng committed:
Merge branch 'branch-A-Packages'

simkianseng committed:
Refactor TaskManager.java to TaskList.java

simkianseng committed:
Merge branch 'add-gradle-support' of https://github.com/SimKianSeng/ip

simkianseng committed:
Add JUnit Java test files

simkianseng committed:
Merge branch 'branch-A-JUnit'

simkianseng committed:
Refactor build.gradle

simkianseng committed:
Add Javadocs

simkianseng committed:
Merge branch 'branch-A-JavaDoc'

simkianseng committed:
Update code according to Java coding standards

simkianseng committed:
Resolve merge conflicts

simkianseng committed:
Add find task by keyword

simkianseng committed:
Resolve merge conflicts

simkianseng committed:
Update Javadoc

simkianseng committed:
Update README.md

kailashgautham commented on src/main/java/simpli/tasks/Event.java:
```diff
@@ -0,0 +1,39 @@
+import java.time.format.DateTimeFormatter;
+
+public class Event extends Task {
+    private static final DateTimeFormatter FORMATTER = DateTimeFormatter.ofPattern("d/M/yyyy HHmm");
```
Could we consider enhancing the readability of the DateTimeFormatter pattern in the Event class? Perhaps using a more explicit pattern like dd/MM/yyyy HH:mm

kailashgautham commented on src/main/java/simpli/tasks/Event.java:
```diff
@@ -0,0 +1,39 @@
+
+public class Event extends Task {
+    private static final DateTimeFormatter FORMATTER = DateTimeFormatter.ofPattern("d/M/yyyy HHmm");
+    protected LocalDateTime from;
```
I noticed that the from and to fields in the Event class are protected. Would making them private and using getters enhance encapsulation?

kailashgautham commented on src/main/java/simpli/interpreter/Interpreter.java:
```diff
@@ -0,0 +1,118 @@
+            break;
+        }
+        case EVENT: {  // creates event task
+            Task addedTask = taskList.addEvent(tokens, dates);
```
Could we consider adding some error handling here to check if the "to" date happens after the "from" date?

kailashgautham commented on src/main/java/simpli/parser/Parser.java:
```diff
@@ -0,0 +1,88 @@
+     * @return An Array of tokens.
+     * @throws ActionException if there are unknown tokens or invalid commands.
+     */
+    public String[] parseCommand(String content) throws ActionException {
```
I like the way this function handles the parsing of the different user inputs. Great job!

kailashgautham reviewed (commented):
Overall, your codebase looks great. Keep it up!

evedaktyl commented on src/main/java/simpli/tasks/Task.java:
```diff
@@ -0,0 +1,59 @@
+
+    /**
+     * Initializes a task with the specified attributes.
+     * @param isDone boolean representing if a task is completed or not.
```
Could we add an empty line between the description and parameter section here similar to the other method headers?

evedaktyl commented on src/test/java/ParserTest.java:
```diff
@@ -0,0 +1,61 @@
+import simpli.exceptions.ActionException;
+import simpli.exceptions.TaskException;
+
+import static org.junit.jupiter.api.Assertions.*;
```
Should these import statements be organised differently? Could we also consider avoiding the use of wildcard imports?

evedaktyl commented on src/main/java/simpli/core/Simpli.java:
```diff
@@ -0,0 +1,110 @@
+            ui.display("The file is corrupted :(");
+        } catch (ActionException e) {
+            ui.display("There is something wrong with the data/simpli.csv file :(");
+        } catch (TaskException e) {
```
I like how you extracted out two kinds of Simpli Exceptions to target unsupported action commands and invalid task parameters separately.

evedaktyl reviewed (commented):
Overall the code is very readable and I particularly liked your input parsing function. Great job on the creative naming as well.

simkianseng committed:
Add JavaFX GUI

simkianseng committed:
Add more JavaFX GUI components

simkianseng committed:
Merge branch 'branch-Level-10'

simkianseng committed:
Refactor code to meet coding standards

simkianseng committed:
Refactor more code to meet coding standards

simkianseng committed:
Merge branch 'branch-A-CheckStyle'

simkianseng committed:
Update JavaDocs

JavaDocs for Storage.java and Gui.java are being updated to meet Git
convention standards.

simkianseng committed:
Add assert statements

Added assert statements to check for important points before data is
being passed to subsequent stages.
Ensured that taskList cannot be null when passed to Interpreter.
Ensured that parser, intrpr, taskList cannot be null when passed to
Storage.
Ensured that the addition of todo, deadline and event tasks incremented
the tasklist size.
Ensured that the deletion of tasks decrement the tasklist size.
Ensured that chatbot response cannot be an empty String.

simkianseng committed:
Merge pull request #2 from SimKianSeng/branch-A-Assertions

Merge branch 'branch-A-Assertions'

simkianseng committed:
Refactor files to improve code quality

Broken down large methods into smaller ones in Interpreter.java. Ensured
that the SLAP is followed.

simkianseng committed:
Merge branch 'master' into branch-A-CodeQuality

simkianseng committed:
Merge pull request #3 from SimKianSeng/branch-A-CodeQuality

Refactor files to improve code quality

simkianseng committed:
Merge branch 'branch-A-CodeQuality' of https://github.com/SimKianSeng/ip into branch-A-CodeQuality

* 'branch-A-CodeQuality' of https://github.com/SimKianSeng/ip:
  Add assert statements
  Update JavaDocs

simkianseng committed:
Merge branch 'master' of https://github.com/SimKianSeng/ip into branch-A-CodeQuality

* 'master' of https://github.com/SimKianSeng/ip:

simkianseng committed:
Add deadline sorting

Tasks added are automatically sorted by deadline tasks first with
deadlines that are approaching being on top. The other tasks go by which
ever task was added first.

simkianseng committed:
Merge branch 'branch-C-sort'

simkianseng committed:
Merge branch 'master' into branch-A-CodeQuality

simkianseng committed:
Merge pull request #4 from SimKianSeng/branch-A-CodeQuality

Refactor code quality 2

simkianseng committed:
Refactor code quality to meet SLAP

simkianseng committed:
Merge branch 'branch-A-CodeQuality'

simkianseng committed:
Merge branch 'master' of https://github.com/SimKianSeng/ip

simkianseng committed:
Update JavaDocs

simkianseng committed:
Update JUnit tests

simkianseng committed:
Update product name in window title

simkianseng committed:
Update dialog box to fit show overflow contents

simkianseng committed:
Update save whenever a new task is added

simkianseng committed:
Update code style

simkianseng committed:
Add user guide

simkianseng committed:
Update user guide

simkianseng committed:
Create gradle.yml

simkianseng committed:
Update gradle.yml

simkianseng committed:
Add GitHub Pages themes

simkianseng committed:
Update readme

simkianseng committed:
Update ignore jekyll vendor files
